Disconnect clients in test framework (#373)
spatialos/UnrealGDKTestGyms: USER_MANUAL.md

##### Player disconnect gym
* Demonstrates that:
  * Players are cleaned up correctly when they disconnect.

> * NOTE: This gym can only be run manually or automatically.

Reviewer: It seems other tests use something like 
```suggestion
 * NOTE: This gym can be run both as an automated test and a manual one. To run it automatically, use [these steps](#how-to-run-the-automated-test-gyms).
```

* Pre-test steps:

>   * In the Unreal Editor's Content Browser, locate `Content/Maps/PlayerDisconnectGym1` and double click to open it.

Reviewer: what is the 2 for?

Author: When you run the tests manually you can run both tests from the same map but as the automated testing does not re-load the map between tests, the clients do not get re-started so you cannot run two automated disconnect tests on the same map. So I duplicated the map to use for automated testing.

Author: I've consolidated this to one map and one test as both tests ultimately tested the same underlying functionality of returning the client to the main menu.

  * Players are cleaned up correctly when they disconnect.

> * NOTE: This gym can only be run manually (This can be automated once the test frameworks supports client disconnects [UNR-529](https://improbableio.atlassian.net/browse/UNR-529)).

Reviewer 2: Can you delete this line now that you've automated this?

> * NOTE: This gym can only be run manually (This can be automated once the test frameworks supports client disconnects [UNR-529](https://improbableio.atlassian.net/browse/UNR-529)).

Reviewer 2: ```suggestion
```

* Pre-test steps:

>   * In the Unreal Editor's Content Browser, locate `Content/Maps/SpatialPlayerDisconnectGym` and double click to open it.

Reviewer 2: I can't find this map in my local instance. I've run test map generation. Am I missing something?

>   * In the Unreal Editor's Content Browser, locate `Content/Maps/SpatialPlayerDisconnectGym` and double click to open it.

Reviewer 2: ```suggestion
  * In the Unreal Editor's Content Browser, locate `Content/Maps/SpatialPlayerDisconnectMap` and double click to open it.
```
